File: src/main/kotlin/com/huawei/cangjie1/psi/CjFunction.java
package com.huawei.cangjie1.psi;


import com.huawei.cangjie1.psi.stubs.CangJieFunctionStub;
import com.huawei.cangjie1.psi.stubs.elements.CjStubElementTypes;
import com.intellij.lang.ASTNode;
import com.intellij.psi.PsiElement;
import com.intellij.psi.PsiModifiableCodeBlock;
import com.intellij.psi.stubs.IStubElementType;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import java.util.List;

public interface CjFunction extends CjDeclarationWithBody, CjCallableDeclaration {
    boolean isLocal();
}

File: src/main/kotlin/com/huawei/cangjie/psi/CjFunction.kt
package com.huawei.cangjie.psi

import com.intellij.psi.PsiElement


interface CjFunction : CjDeclarationWithBody, CjCallableDeclaration {
    val isLocal: Boolean
    val isStatic: Boolean
        get() = false
    val isOperator: Boolean
        get() = false
    val isMut get() = false

    val isConst get() =  false

    val keyword :PsiElement? get() = null
}

